### PR TITLE
egress test: fix race conditions, add an explicit timeout

### PR DESCRIPTION
**Pull request by aermakov-zalando**
* create the pod with RestartPolicy: Never, otherwise the test might never capture the exit status
 * time out after 10 minutes if we failed

### CODE REVIEW

File: test/e2e/egress.go [hunk 1]
```diff
@@ -81,37 +82,37 @@ var _ = framework.KubeDescribe("Static Egress creation", func() {
...
-			switch n := p.Status.ContainerStatuses[0].State.Terminated.ExitCode; n {
-			case 0:
-				return
-			case 2:
-				// set error
-				Expect(fmt.Errorf("failed to change public IP")).NotTo(HaveOccurred())
-				return
-			}
-		}
+		Eventually(func() error {
```
Comment (szuecs) [inline test/e2e/egress.go:92]: Does `Eventually` retry?

I think the problem was that it takes a while to actually have everything setup before you get the routing change applied.

Comment (aermakov-zalando) [reply test/e2e/egress.go:92]: Yes, it'll retry every 10 seconds and time out after 10 minutes (see https://github.com/zalando-incubator/kubernetes-on-aws/pull/2385/files#diff-b9184f93ce39a9ec0e9abba3438a9650R94).

**Comment by aermakov-zalando** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**